src/planner_test.cc

```diff
@@ -214,28 +246,23 @@ static void VerifySegmentConstraints(const MachineControlConfig &config,
     const GCodeParserAxis axis = (GCodeParserAxis)i;
     const auto steps = segment.steps[axis];
     if (steps == 0) continue;
-    const double accel = fabs_accel(segment.v0, segment.v1, steps);
+    const double factor = get_speed_factor_for_axis(&segment, i);
+    const double v1 = factor * segment.v0;
+    const double v2 = factor * segment.v1;
+    const double accel = fabs_accel(v1, v2, steps);
     const double max_accel =
       config.acceleration[axis] * config.steps_per_mm[axis];
     const double max_feedrate =
       config.max_feedrate[axis] * config.steps_per_mm[axis];
     EXPECT_LE(segment.v0, max_feedrate);
     EXPECT_LE(segment.v1, max_feedrate);
```

Comment:
This doesn't look like a proper check. We should test (once) that the euclidean speed is indeed below the requested feedrate. Nothing else.